src/llmq/quorums_blockprocessor.cpp

```diff
@@ -282,19 +282,19 @@ bool CQuorumBlockProcessor::IsCommitmentRequired(Consensus::LLMQType llmqType, c
         LOCK(cs_main);
         const auto& consensus = Params().GetConsensus();
         if (consensus.nTemporaryTestnetForkDIP3Height != 0 &&
-            pindexPrev->nHeight + 1 > consensus.nTemporaryTestnetForkDIP3Height &&
-            pindexPrev->nHeight + 1 < consensus.nTemporaryTestnetForkHeight &&
+            pindex->nHeight > consensus.nTemporaryTestnetForkDIP3Height &&
+            pindex->nHeight < consensus.nTemporaryTestnetForkHeight &&
             chainActive[consensus.nTemporaryTestnetForkDIP3Height]->GetBlockHash() == consensus.nTemporaryTestnetForkDIP3BlockHash) {
             allowMissingQc = true;
         }
     }
     // END TEMPORARY CODE
 
-    uint256 quorumHash = GetQuorumBlockHash(llmqType, pindexPrev);
+    uint256 quorumHash = GetQuorumBlockHash(llmqType, pindex->pprev);
```

Comment:
Now that we have `pindex` instead of `pindexPrev`, shouldn't we also pass `pindex` here and modify the logic in `GetQuorumBlockHash`? This way we can also remove that special handling for `uint256()` for the first block of a quorum.

Reply:
should be fixed but I'm not sure if it's a "refactoring" anymore 🙈

Reply:
Well it's still a refactoring as it doesn't change any behavior. The `uint256()` case could only happen on the first block of quorum interval, but I simply ignored it by adding a few ifs after calling `GetQuorumBlockHash`. And ignoring those didn't change anything as nothing special is supposed to happen on the first block.